bot/resources/tags/in-place.md

```diff
@@ -0,0 +1,17 @@
+**Out of Place** and **In Place**
+
+In programming, there are two types of operations: "out of place" operations creates a new object, leaving the original object unchanged. "in place" operations modifies the original object, without creating a new one. These operations return None explicitly.
+
+A common example of these different concepts is seen in the use of the methods `list.sort()` and `sorted(...)`. Using `list.sort()` and attempting to access an element of the list will result in an error. 
+
+```py
+a_list = [3, 1, 2]
+a_new_list = a_list.sort() # This will be None
+print(a_new_list[1]) # This will error because it is NoneType and not a list
+
+a_list = [3, 1, 2]
+sorted(a_list)
+print(a_list[0]) # You may expect 1, but it will print 3
+```
+
+To avoid these errors and unexpected results, it is required to assign the result of `sorted(...)` to a new variable and use `list.sort()` method in the original list. This way, the original list will be sorted and the new list will be created with the sorted elements.
```

Comment:
I'm not sure how I like "it is required" here, as well as the focus on lists. My proposal here would something along the lines of:

> To avoid these errors and unexpected results, you should either use an in-place operation (`sorted(...)`) and assign it to a variable or use an out-of-place operation (`list.sort()`) without assignment.

The wording can obviously vary, but I'd prefer to see something general like that.

Reply:
Seems good to me, I chose required because I couldn't really think of anything else in the moment.

Reply:
I think variable names here should be more in-line with what the person was trying to do, so I would call them `unsorted_list` and `sorted_list` for the first example and `list_to_sort` in the second one.

I also think the way you show the unexpected behavior (with indexing) is overcomplicated, something like
```py
print(a_new_list)  # Outputs None. Where did the list go?
```
and
```py
print(outofplace_list)  # The list still isn't sorted. Why?
```
would make it easier to understand for beginners.

Reply:
@n0Oo0Oo0b Seems like a good idea, updated.